connection/src/thick.rs

```diff
@@ -164,7 +170,9 @@ impl AttestedConnection for ThickClient {
         let init_input = ClientInitiate::<X25519, Aes256Gcm, Sha512>::default();
         let (initiator, auth_request_output) = initiator.try_next(&mut csprng, init_input)?;
 
-        let auth_response_msg = self.attested_api_client.auth(&auth_request_output.into())?;
+        let auth_response_msg = self
+            .attested_api_client
+            .auth_opt(&auth_request_output.into(), self.creds.call_option()?)?;
```

Comment:
Passing the `call_option()` returned from `self.creds` is what sticks the authentication HTTP header in the GRPC request.